src/core/connection.c

```diff
@@ -182,7 +182,7 @@ QuicConnAlloc(
 
         Connection->Stats.QuicVersion = Packet->Invariant->LONG_HDR.Version;
         QuicConnOnQuicVersionSet(Connection);
-        QuicCopyRouteInfo(&Path->Route, Packet->Route);
+        QuicCopyRouteInfo(&Path->Route, Packet->Route, ((CXPLAT_RECV_DATA*)Packet)->DatapathType);
```

Comment:
Why don't you save the type in the `Route` if necessary? We should do everything we can to simplify things for the upper layer. It shouldn't have to care about this stuff as much as possible.